python/lsst/ip/diffim/dcrModel.py

```diff
@@ -558,50 +562,46 @@ def applyImageThresholds(self, image, highThreshold=None, lowThreshold=None, reg
             image[lowPixels] = lowThreshold[lowPixels]
 
 
-def applyDcr(maskedImage, dcr, warpCtrl, bbox=None, useInverse=False, splitSubfilters=False):
+def applyDcr(image, dcr, useInverse=False, splitSubfilters=False, **kwargs):
     """Shift a masked image.
 
     Parameters
     ----------
-    maskedImage : `lsst.afw.image.MaskedImage`
-        The input masked image to shift.
-    dcr : `lsst.afw.geom.Extent2I`
+    image : `numpy.ndarray`
+        The input image to shift.
+    dcr : `tuple` of two `float`
         Shift calculated with ``calculateDcr``.
-    warpCtrl : `lsst.afw.math.WarpingControl`
-        Configuration settings for warping an image
-    bbox : `lsst.afw.geom.Box2I`, optional
-        Sub-region of the masked image to shift.
-        Shifts the entire image if None (Default).
+        Uses numpy axes ordering (Y, X).
     useInverse : `bool`, optional
         Use the reverse of ``dcr`` for the shift. Default: False
     splitSubfilters : `bool`, optional
         Calculate DCR for two evenly-spaced wavelengths in each subfilter,
         instead of at the midpoint. Default: False
+    kwargs
+        Additional keyword parameters to pass in to
+        `scipy.ndimage.interpolation.shift`
 
     Returns
     -------
-    shiftedImage : `lsst.afw.image.maskedImageF`
-        A masked image, with the pixels within the bounding box shifted.
+    shiftedImage : `numpy.ndarray`
+        A copy of the input image with the specified shift applied.
     """
-    padValue = afwImage.pixel.SinglePixelF(0., maskedImage.mask.getPlaneBitMask("NO_DATA"), 0)
-    if bbox is None:
-        bbox = maskedImage.getBBox()
     if splitSubfilters:
-        shiftedImage = afwImage.MaskedImageF(bbox)
-        transform0 = makeTransform(AffineTransform((-1.0 if useInverse else 1.0)*dcr[0]))
-        afwMath.warpImage(shiftedImage, maskedImage[bbox],
-                          transform0, warpCtrl, padValue=padValue)
-        shiftedImage1 = afwImage.MaskedImageF(bbox)
-        transform1 = makeTransform(AffineTransform((-1.0 if useInverse else 1.0)*dcr[1]))
-        afwMath.warpImage(shiftedImage1, maskedImage[bbox],
-                          transform1, warpCtrl, padValue=padValue)
-        shiftedImage += shiftedImage1
+        if useInverse:
+            shift = [-1.*s for s in dcr[0]]
+            shift1 = [-1.*s for s in dcr[1]]
```

Comment:
The docstring says `dcr` is a `tuple of two float`, so how can this work?

Reply:
I'll fix the docstring. `dcr` has a different form if `splitSubfilters` is set.